pallas-txbuilder/src/transaction/mod.rs

```diff
@@ -0,0 +1,50 @@
+use minicbor::{Decode, Encode};
+use pallas_crypto::hash::Hash;
+use pallas_primitives::{
+    babbage::{AuxiliaryData, TransactionBody, TransactionInput, WitnessSet},
+    Fragment,
+};
+
+mod output;
+
+pub use output::*;
+
+#[derive(Debug, Clone)]
+pub struct Input {
+    transaction_id: Hash<32>,
+    index: u64,
+}
+
+impl Input {
+    pub fn new(transaction_id: impl Into<Hash<32>>, index: u64) -> Self {
+        Self {
+            transaction_id: transaction_id.into(),
+            index,
+        }
+    }
+
+    pub fn build(self) -> TransactionInput {
+        TransactionInput {
+            transaction_id: self.transaction_id,
+            index: self.index,
+        }
+    }
+}
+
+#[derive(Debug, Encode, Decode, Clone)]
+pub struct Transaction {
```

Comment:
Was it a pain or not reasonable to reuse the one defined in primitives?

Reply:
I'm going to check again, but afaik there's only `TransactionBody` in primitives, not `Transaction` itself.

Reply:
Oh right that's correct

Reply:
actually turns out it's right here

https://github.com/cfcosta/pallas/blob/ea9c4e8b59fe5cbaaf73c649b44347a066f70960/pallas-primitives/src/babbage/model.rs#L699